src/coreHandler.ts

```diff
@@ -128,37 +130,24 @@ export class CoreHandler {
 	 * Get options for connecting to core
 	 */
 	getCoreConnectionOptions(deviceOptions: DeviceConfig, name: string): CoreOptions {
-		let credentials: {
-			deviceId: string
-			deviceToken: string
-		}
-
-		if (deviceOptions.deviceId && deviceOptions.deviceToken) {
-			credentials = {
-				deviceId: deviceOptions.deviceId,
-				deviceToken: deviceOptions.deviceToken,
-			}
-		} else if (deviceOptions.deviceId) {
-			this.logger.warn('Token not set, only id! This might be unsecure!')
-			credentials = {
-				deviceId: deviceOptions.deviceId + name,
-				deviceToken: 'unsecureToken',
-			}
-		} else {
-			credentials = CoreConnection.getCredentials(name.replace(/ /g, ''))
-		}
 		let options: CoreOptions = {
-			...credentials,
-
-			deviceCategory: P.DeviceCategory.INGEST,
-			deviceType: P.DeviceType.INEWS,
-			deviceSubType: P.SUBTYPE_PROCESS,
+			deviceId: protectString(deviceOptions.deviceId + !deviceOptions.deviceToken ? name : ''),
```

Comment:
I think it could look like this:
```suggestion
			deviceId: protectString(deviceOptions.deviceId + subDeviceId),
```
where `subDeviceId` is a string without whitespaces. See the [MOS Gateway](https://github.com/tv2/tv-automation-server-core/blob/7069680c1880591ccb6dfe56426112f5d47ce3c2/packages/mos-gateway/src/coreHandler.ts#L642-L649)

Reply:
With this change, i take that `subDeviceId` would be an additional parameter for the containing method. It only looks like `getCoreConnectionOptions(...)` gets called ones, so it thankfully wouldn't be hard to update the callsites.

What should this new parameter be set to however? Should it just say 'iNewsGateway'? like it currently does in `name`, simple with a space in the string. If so how much is there to gain over e.g trimming all spaces from the `name` parameter? I see that for the MOS Gateway, the `subDeviceId` is set to 'MosCoreParent'.

Reply:
Yup, as far as I know, our iNews Gateway does not have any child devices, so this does not have to be an additional parameter because indeed we're calling `getCoreConnectionOptions` once. I'm fine with hardcoding it to `iNewsGateway` or removing whitespaces from `name`. The only importand requirement is that it has no whitespaces, because although unlikely, it seems like they may cause issues being in an ID that we're using in places like URLs etc.

Reply:
Ok, as it sound like it would contain the exact same value, minus the space, i will make it create a spaceless value from name. Hmm, at the moment it is only added when the token is null/undefined, should that still be the case or always add it?